code/modules/assembly/holder.dm

```diff
@@ -62,54 +71,46 @@
 
 /obj/item/assembly_holder/update_overlays()
 	. = ..()
-	if(a_left)
-		. += "[a_left.icon_state]_left"
-		for(var/left_overlay in a_left.attached_overlays)
-			. += "[left_overlay]_l"
-
-	if(!a_right)
-		return
-
-	if(a_right.is_position_sensitive)
-		. += "[a_right.icon_state]_right"
-		for(var/right_overlay in a_right.attached_overlays)
-			. += "[right_overlay]_r"
-		return
-
-	var/mutable_appearance/right = mutable_appearance(icon, "[a_right.icon_state]_left")
-	right.transform = matrix(-1, 0, 0, 0, 1, 0)
-	for(var/right_overlay in a_right.attached_overlays)
-		right.add_overlay("[right_overlay]_l")
-	. += right
+	for(var/i in 1 to LAZYLEN(assemblies))
+		if(i % 2 == 1)
+			var/obj/item/assembly/assembly = assemblies[i]
+			. += "[assembly.icon_state]_left"
+			for(var/left_overlay in assembly.attached_overlays)
+				. += "[left_overlay]_l"
+		if(i % 2 == 0)
+			var/obj/item/assembly/assembly = assemblies[i]
+			var/mutable_appearance/right = mutable_appearance(icon, "[assembly.icon_state]_left")
+			right.transform = matrix(-1, 0, 0, 0, 1, 0)
+			for(var/right_overlay in assembly.attached_overlays)
+				right.add_overlay("[right_overlay]_l")
+			. += right
 
 /obj/item/assembly_holder/on_found(mob/finder)
-	if(a_left)
-		a_left.on_found(finder)
-	if(a_right)
-		a_right.on_found(finder)
+	for(var/obj/item/assembly/assembly as anything in assemblies)
+		assembly.on_found(finder)
 
 /obj/item/assembly_holder/setDir()
 	. = ..()
-	if(a_left)
-		a_left.holder_movement()
-	if(a_right)
-		a_right.holder_movement()
+	for(var/obj/item/assembly/assembly as anything in assemblies)
+		assembly.holder_movement()
 
 /obj/item/assembly_holder/dropped(mob/user)
 	. = ..()
-	if(a_left)
-		a_left.dropped()
-	if(a_right)
-		a_right.dropped()
+	for(var/obj/item/assembly/assembly as anything in assemblies)
+		assembly.dropped()
 
 /obj/item/assembly_holder/attack_hand(mob/living/user, list/modifiers)//Perhapse this should be a holder_pickup proc instead, can add if needbe I guess
 	. = ..()
 	if(.)
 		return
-	if(a_left)
-		a_left.attack_hand()
-	if(a_right)
-		a_right.attack_hand()
+	for(var/obj/item/assembly/assembly as anything in assemblies)
+		assembly.attack_hand()
+
+/obj/item/assembly_holder/attackby(obj/item/weapon, mob/user, params)
+	if(isassembly(weapon))
+		var/obj/item/assembly/attached_assembly = weapon
+		if(!attached_assembly.secured)
+			add_assembly(attached_assembly,user)
```

Comment:
```suggestion
/obj/item/assembly_holder/attackby(obj/item/weapon, mob/user, params)
	if(isassembly(weapon))
		var/obj/item/assembly/attached_assembly = weapon
		if(!attached_assembly.secured)
			add_assembly(attached_assembly,user)
		return
	return ..()
```
`attackby()` should probably call parent if it's a non-assembly item.
Bonus points: Give a balloon alert if someone tries to attach an assembly that isn't secured